go/kex2/provisioner.go

```diff
@@ -118,9 +118,10 @@ func (p *provisioner) pickFirstConnection() (err error) {
 		if err = srv.Register(prot); err != nil {
 			return err
 		}
-		if err = srv.Run(true); err != nil {
-			return err
-		}
+		go func() {
+			_ = srv.Run()
+			// TODO: do something with the error.
```

Comment:
Would it be easy to log it now, instead of doing nothing? Or maybe if the error is never supposed to happen, panic?